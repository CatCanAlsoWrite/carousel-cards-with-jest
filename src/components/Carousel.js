"use client";

import React, { useState } from "react";
import Card from "../components/Card";
import Button from "../components/Button";

const Carousel = ({ data }) => {
  const [slide, setSlide] = useState(3);

  const prevSlide = () => {
    setSlide(0);
  };
  const nextSlide = () => {
    setSlide(3);
  };

  return (
    <div>
      <div className="overflow-hidden flex  font-[next-poster]">
        {data.map((card, index) => {
          return (
            <div key={index} className={`card${card.id} relative pt-[4px]`}>
              {slide === index || <Card card={card} />}
            </div>
          );
        })}
      </div>
      <div className="w-[1602px] relative ">
        {slide === 0 || (
          <Button
            onClick={prevSlide}
            position="right-0"
            bg="bg-gradient-to-l"
            margin="ml-[18px]"
            img="M1 1L12 9L1 17"
          />
        )}
        {slide === 3 || (
          <Button
            onClick={nextSlide}
            position="left-0"
            bg="bg-gradient-to-r"
            margin="ml-[121px]"
            img="M12 17L1 9L12 1"
          />
        )}
      </div>
    </div>
  );
};

export default Carousel;
