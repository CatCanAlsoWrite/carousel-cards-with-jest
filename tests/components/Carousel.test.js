import React from "react";
import { render } from "@testing-library/react";
import Carousel from "../../src/components/Carousel";

const data = [
  { id: 1, title: "slide 1", content: "slide 1" },
  { id: 2, title: "slide 2", content: "slide 2" },
  { id: 3, title: "slide 3", content: "slide 3" },
  { id: 4, title: "slide 4", content: "slide 4" },
];

describe("Carousel", () => {
  it("can show slide data", () => {
    const { baseElement } = render(<Carousel data={data} />);
    expect(baseElement).toMatchSnapshot();
  });
});
