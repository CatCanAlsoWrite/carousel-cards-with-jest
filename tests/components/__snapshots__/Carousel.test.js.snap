// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel can show slide data 1`] = `
<body>
  <div>
    <div>
      <div
        class="overflow-hidden flex  font-[next-poster]"
      >
        <div
          class="card1 relative pt-[4px]"
        />
        <div
          class="card2 relative pt-[4px]"
        >
          <div>
            <div
              class="invisible mb-[40px]"
            >
              &lt; / Trending Now &gt;
            </div>
            <div
              class="cardHoverBgImg2 absolute left-0 bottom-0 card w-[416px] h-[278px]"
            />
            <div
              class="cardBgImg2 card w-[416px] h-[278px] transition-all duration-300 hover:translate-y-[-4px] hover:translate-x-[4px]"
            >
              <div
                class="mx-[40px]"
              >
                <div
                  class="cardBar2 w-[46px] h-[4px] rounded-[11px] "
                />
                <div
                  class="undefined card-title font-[next-book-bold] text-[16px] leading-[16px]"
                />
                <p
                  class="text-[12px] leading-[13.2px] text-[color:var(--card-gray-hex)]"
                >
                  slide 2
                </p>
                <div
                  class="cardBottom2 flex justify-between items-center"
                >
                  <div
                    class="flex gap-[12px] font-[neue-machina] text-[12px] leading-[12px]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="card3 relative pt-[4px]"
        >
          <div>
            <div
              class="invisible mb-[40px]"
            >
              &lt; / Trending Now &gt;
            </div>
            <div
              class="cardHoverBgImg3 absolute left-0 bottom-0 card w-[416px] h-[278px]"
            />
            <div
              class="cardBgImg3 card w-[416px] h-[278px] transition-all duration-300 hover:translate-y-[-4px] hover:translate-x-[4px]"
            >
              <div
                class="mx-[40px]"
              >
                <div
                  class="cardBar3 w-[46px] h-[4px] rounded-[11px] "
                />
                <div
                  class="undefined card-title font-[next-book-bold] text-[16px] leading-[16px]"
                />
                <p
                  class="text-[12px] leading-[13.2px] text-[color:var(--card-gray-hex)]"
                >
                  slide 3
                </p>
                <div
                  class="cardBottom3 flex justify-between items-center"
                >
                  <div
                    class="flex gap-[12px] font-[neue-machina] text-[12px] leading-[12px]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="card4 relative pt-[4px]"
        >
          <div>
            <div
              class="invisible mb-[40px]"
            >
              &lt; / Trending Now &gt;
            </div>
            <div
              class="cardHoverBgImg4 absolute left-0 bottom-0 card w-[416px] h-[278px]"
            />
            <div
              class="cardBgImg4 card w-[416px] h-[278px] transition-all duration-300 hover:translate-y-[-4px] hover:translate-x-[4px]"
            >
              <div
                class="mx-[40px]"
              >
                <div
                  class="cardBar4 w-[46px] h-[4px] rounded-[11px] "
                />
                <div
                  class="undefined card-title font-[next-book-bold] text-[16px] leading-[16px]"
                />
                <p
                  class="text-[12px] leading-[13.2px] text-[color:var(--card-gray-hex)]"
                >
                  slide 4
                </p>
                <div
                  class="cardBottom4 flex justify-between items-center"
                >
                  <div
                    class="flex gap-[12px] font-[neue-machina] text-[12px] leading-[12px]"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-[1602px] relative "
      >
        <div
          class="left-0 bg-gradient-to-r flex items-center absolute top-[-278px] w-[192px] h-[278px] pointer-events-auto  from-[color:var(--cover-arrow-hex)] via-[color:var(--cover-arrow-via-hex)]"
        >
          <button
            class="ml-[121px] btn btn-circle border-[color:var(--card-gray-hex)] hover:border-0 bg-transparent hover:bg-[color:var(--card-gray-hex)]"
          >
            <svg
              fill="none"
              height="18"
              viewBox="0 0 13 18"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17L1 9L12 1"
                stroke="white"
                stroke-linejoin="bevel"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
